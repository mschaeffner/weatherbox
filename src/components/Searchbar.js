import React from 'react'
import Autocomplete from 'react-autocomplete';


const renderSearchResultItem = (place, isHighlighted) =>
  <div style={{ padding: '5px', background: isHighlighted ? '#EEE' : 'white' }}>
    {place.name + ' (' + place.country + ')'}
  </div>

export default ({keyword, places, onSearch, onSelect}) =>
  <div style={{position: 'absolute', zIndex:100}}>
    <Autocomplete
      inputProps={{style:styles.input, placeholder:'Search for a place' }}
      menuStyle={styles.input}
      getItemValue={(place) => place.id + ''}
      items={places}
      renderItem={(place, isHighlighted) => renderSearchResultItem(place, isHighlighted)}
      value={keyword}
      onChange={event => onSearch(event.target.value)}
      onSelect={(value, place) => onSelect(place)}
    />
  </div>

const styles = {
  input: {
    fontSize: '16px',
    padding: '5px',
    width: '300px'
  }
}
