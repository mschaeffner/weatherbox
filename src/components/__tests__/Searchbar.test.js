import React from 'react';
import renderer from 'react-test-renderer';
import Searchbar from '../Searchbar';


describe('Searchbar', () => {
  it('should show searchbar and suggestions based on current keyword', () => {

    const places = [
      { id: 123, name: 'Berlin', country: 'DE' },
      { id: 456, name: 'Bernau', country: 'DE' },
      { id: 789, name: 'Berghagen', country: 'DE' }
    ]

    const component = renderer.create(<Searchbar keyword="Ber" places={places} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
