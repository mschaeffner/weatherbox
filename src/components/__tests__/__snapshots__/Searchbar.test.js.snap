// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbar should show searchbar and suggestions based on current keyword 1`] = `
<div
  style={
    Object {
      "position": "absolute",
      "zIndex": 100,
    }
  }
>
  <div
    style={
      Object {
        "display": "inline-block",
      }
    }
  >
    <input
      aria-autocomplete="list"
      aria-expanded={false}
      autoComplete="off"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      placeholder="Search for a place"
      role="combobox"
      style={
        Object {
          "fontSize": "16px",
          "padding": "5px",
          "width": "300px",
        }
      }
      value="Ber"
    />
  </div>
</div>
`;
